### PR TITLE
Switch from GLEW to epoxy

### PR DESCRIPTION
**Pull request by JayFoxRox**
This implements epoxy as backend for glo (which used to use GLEW).
As epoxy does not do context management, I decided to stick to glo (QEMU uses EGL exclusively)

I did not integrate epoxy directly in our codebase yet due to unfamiliarity with their API, and because I didn't want to delay fixing master.

The display flickers with Gtk, but works fine with SDL.
As I can't run XQEMU without this patch, I don't know if this change is responsible for that problem or not.

---

As I'm almost exclusively Linux (GLX) user. Therefore I depend on test results by macOS (CGL) and Windows (WGL) users.
Please confirm that XQEMU compiles and runs fine on those platforms with these changes.

---

TODO:

- Wait for github support to switch upstream to xqemu/xqemu
- Have people test on windows and macOS

### CODE REVIEW

File: hw/xbox/nv2a/gl/gloffscreen_common.c [hunk 1]
```diff
@@ -77,15 +77,7 @@ void glo_readpixels(GLenum gl_format, GLenum gl_type,
...
 {
-    int i;
-    int num_extensions = GL_NUM_EXTENSIONS;
-    for (i=0; i<num_extensions; i++) {
-      const char* ext = (const char*)glGetStringi(GL_EXTENSIONS, i);
-      if (!ext) break;
-      if (strcmp(ext, ext_name) == 0) return true;
-    }
-    return false;
+  return epoxy_has_gl_extension(ext_name);
```
Comment (JayFoxRox) [inline hw/xbox/nv2a/gl/gloffscreen_common.c]: indentation is bad

File: hw/xbox/nv2a/nv2a_debug.c [hunk 2]
```diff
@@ -87,9 +107,16 @@ void gl_debug_label(GLenum target, GLuint name, const char *fmt, ...)
...
     va_end(ap);
 
-    if (glObjectLabel) {
-        glObjectLabel(target, name, n, buffer);
+    glObjectLabel(target, name, n, buffer);
+}
+
+void gl_debug_frame_terminator(void)
+{
+    if (has_GL_GREMEDY_frame_terminator) {
```
Comment (JayFoxRox) [inline hw/xbox/nv2a/nv2a_debug.c]: check is negated

File: hw/xbox/nv2a/nv2a_debug.c [hunk 3]
```diff
@@ -51,18 +65,19 @@ void gl_debug_message(bool cc, const char *fmt, ...)
...
-    size_t n;
-    char buffer[1024];
-    va_list ap;
-    va_start(ap, fmt);
-    n = vsnprintf(buffer, sizeof(buffer), fmt, ap);
-    assert(n <= sizeof(buffer));
-    va_end(ap);
-
     /* Check for errors before entering group */
     assert(glGetError() == GL_NO_ERROR);
```
Comment (JayFoxRox) [inline hw/xbox/nv2a/nv2a_debug.c]: Could be part of the debug group already?

**Comment by illwieckz** [comment]
It works on my end, obsoleting xqemu/xqemu#13 and fixing xqemu/xqemu#53 and xqemu/xqemu#20

I tested it on this system:

```
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04 LTS
Release:	18.04
Codename:	bionic

$ uname -a
Linux gollum 4.16.0-16.2-liquorix-amd64 #1 ZEN SMP PREEMPT liquorix 4.16-9.1~sid (2018-06-19) x86_64 x86_64 x86_64 GNU/Linux

$ glxinfo | grep OpenGL
OpenGL vendor string: X.Org
OpenGL renderer string: AMD Radeon (TM) R9 390 Series (HAWAII, DRM 3.23.0, 4.16.0-16.2-liquorix-amd64, LLVM 6.0.0)
OpenGL core profile version string: 4.5 (Core Profile) Mesa 18.1.1 - padoka PPA
OpenGL core profile shading language version string: 4.50
OpenGL core profile context flags: (none)
OpenGL core profile profile mask: core profile
OpenGL core profile extensions:
OpenGL version string: 3.1 Mesa 18.1.1 - padoka PPA
OpenGL shading language version string: 1.40
OpenGL context flags: (none)
OpenGL extensions:
OpenGL ES profile version string: OpenGL ES 3.1 Mesa 18.1.1 - padoka PPA
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.10

$ dpkg -l | grep 'mesa.*-dev:\|epoxy.*-dev:'
ii  libegl1-mesa-dev:amd64                        18.1.1-0~b~padoka0                                                          amd64        free implementation of the EGL API -- development files
ii  libepoxy-dev:amd64                            1.4.3-1                                                                     amd64        OpenGL function pointer management library- development
ii  libgl1-mesa-dev:amd64                         18.1.1-0~b~padoka0                                                          amd64        free implementation of the OpenGL API -- GLX development files
ii  libgles2-mesa-dev:amd64                       18.1.1-0~b~padoka0                                                          amd64        free implementation of the OpenGL|ES 2.x API -- development files
ii  libglu1-mesa-dev:amd64                        9.0.0-2.1build1                                                             amd64        Mesa OpenGL utility library -- development files
ii  mesa-common-dev:amd64                         18.1.1-0~b~padoka0                                                          amd64        Developer documentation for Mesa
```

**Comment by mborgerson** [comment]
Please add "-lepoxy" on line ./configure line 2951 such that:

```
if test "$darwin" = "yes" ; then
    sdl_libs="$sdl_libs -framework OpenGL -lepoxy"
...
```

**Comment by JayFoxRox** [comment]
* Added `-lepoxy` as requestd by @mborgerson 
* Split debugging into #30 because I don't currently have time to test it and it's really a seperate change (I'm fine with merging it into this one if users report problems, otherwise I'll send it shortly after this one)
* Also running CI now (which is slightly misconfigured)

**Comment by mborgerson** [comment]
Appears to be crashing in Windows for unknown reasons. Needs further investigation before merging.

**Comment by JayFoxRox** [comment]
Moved to https://github.com/JayFoxRox/xqemu/pull/1 to kickoff another CI.
I'll integrate all reports from this one in the new repo and make sure the existing review / testing is not forgotten in upstream PR.

**Pull request closed without merging**